projects/saturn/src/main/java/io/fairspace/saturn/webdav/DirectoryResource.java

```diff
@@ -75,12 +76,36 @@ public Date getModifiedDate() {
         // create directory
         var subj = createResource(newName).addProperty(RDF.type, FS.Directory);
 
-        // create entity
-        createLinkedEntity(newName, subj, factory);
+        //create linked entity
+        var type = factory.getLinkedEntityType();
+        validateLinkedEntityType(subj, type);
+        factory.createLinkedEntity(newName, subj, type);
 
         return (io.milton.resource.CollectionResource) factory.getResource(subj, access);
     }
 
+    private void validateLinkedEntityType(org.apache.jena.rdf.model.Resource resource, org.apache.jena.rdf.model.Resource type) throws BadRequestException {
+        var parentType = Optional.ofNullable(resource.getPropertyResourceValue(FS.belongsTo))
```

Comment:
I would expect to follow the 'linkedEntity' path, does this work with 'belongsTo'?

Reply:
"belongsTo" is a link between directories, which is needed here, e.g. PI_directory belongs to Department_directory. Linked entity is just the linked to metadata entity, e.g. PI_directory to PI entity.